### PR TITLE
fix: clear ContextsManager timers when app exits

### LINKED ISSUE #6198: Timers are causing uncaught exceptions when exiting
The `ContextsManager` (`packages/main/src/plugin/kubernetes-context-state.ts`) allocates timers during its lifecycle, but these timers are not disposed when Podman Desktop exits, and this can lead to such error messages when exiting from Podman Desktop:

timer-error

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Clear ContextsManager timers when app exits.

### What issues does this PR fix or reference?

Fixes #6198

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/main/src/plugin/index.ts [hunk 1]
```diff
@@ -181,7 +181,7 @@ export class PluginSystem {
   private uiReady = false;
 
   // true if the application is quitting
-  private isQuitting = false;
+  public isQuitting = false;
```
Comment (benoitf) [inline packages/main/src/plugin/index.ts]: I think that it needs to stay private

if you need the value, you can add an accessor but we can't let anyone being able to override the value

**Comment by benoitf** [comment]
I think it's a workaround correct ? as the root cause is that timers should be canceled

**Comment by benoitf** [comment]
I don't see how it's fixing the root problem/issue (It's just hiding the problem)

**Comment by feloy** [comment]
Yes, it's a workaround, as I didn't find the way to stop the timers when the app was exiting. But working on this side, it helped me realize that the kubernetes-client can be informed to stop the timers when this event is triggered: 
```
app.on('before-quit', () => {
  this.isQuitting = true;
});
```

I'll make the changes

**Comment by benoitf** [comment]
please do not add dependency to app events from anything in the plugin/** part

all objects created during the start of the plug-in context should receive teardown by the same calls that created them

**Comment by feloy** [comment]
> please do not add dependency to app events from anything in the plugin/** part
> 
> all objects created during the start of the plug-in context should receive teardown by the same calls that created them

I tried to apply this. Please tell me if it's ok

File: packages/main/src/plugin/index.ts [hunk 2]
```diff
@@ -459,6 +459,10 @@ export class PluginSystem {
 
     const messageBox = new MessageBox(apiSender);
 
+    app.on('before-quit', () => {
```
Comment (benoitf) [inline packages/main/src/plugin/index.ts]: we've already this event handled in this class and we call extension loader stop all extensions

For now, let's move the kubernetesClient dispose to extension loader.stopAllExtensions

We'll add later a stop all that should call stop all extensions and dispose on the kubernetesClient

**Comment by feloy** [comment]
I still need to add some tests on the dispose() method

**Comment by benoitf** [comment]
thanks @feloy , works for me

**Review by deboer-tim — APPROVED** [review]
I'm generally not able to reproduce the original problem, but testing on this PR caused no issues and code LGTM.

**Comment by benoitf** [comment]
created https://github.com/containers/podman-desktop/issues/6288 as follow-up issue

**Review by benoitf — APPROVED** [review]

**Pull request merged**